realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsQueryTests.java

```diff
@@ -335,41 +329,36 @@ public void isNotNull_list() {
 
     @Test
     public void isNotNull_unsupported() {
-        long result;
-
         // Tests for other unsupported not null types are in RealmQueryTests
 
         try {
-            result = realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_LO_OBJECT).count();
+            realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_LO_OBJECT).count();
             fail("isNotNull should throw on type LINKING_OBJECT(14) targeting an OBJECT");
         } catch (IllegalArgumentException expected) {
-            assertEquals("Illegal Argument: LinkingObject from field fieldObjectNull is not nullable.", expected.getMessage());
+            assertTrue(expected.getMessage().contains("Illegal Argument: Cannot compare linklist ('@links.class_NullTypes.fieldObjectNull')"));
         }
         try {
-            result = realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_LO_LIST).count();
+            realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_LO_LIST).count();
             fail("isNotNull should throw on type LINKING_OBJECT(14) targeting a LIST");
         } catch (IllegalArgumentException expected) {
-            assertEquals("Illegal Argument: LinkingObject from field fieldListNull is not nullable.", expected.getMessage());
+            assertTrue(expected.getMessage().contains("Illegal Argument: Cannot compare linklist ('@links.class_NullTypes.fieldListNull')"));
         }
     }
 
     @Test
     public void isNotNull_unsupportedLinkedTypes() {
-        RealmQuery<NullTypes> result;
-
         // Tests for other unsupported not null types are in RealmQueryTests
-
         try {
-            result = realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_LO_OBJECT);
+            realm.where(NullTypes.class).isNotNull(NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_LO_OBJECT).count();
```

Comment:
Was the count missing before, or has something changed in the test? Or are we changing semantics?

Similar in other tests!?

Reply:
Good catch, completely forgot to remove those.